### ISSUE TITLE
ROS2 support, and raspberry pi?

### ISSUE THREAD

**Issue by sequoia-hope** [alpine-ros/alpine-ros#20]
I apologize if this has been answered elsewhere. Is there any plan to support ROS2? Is there anything I could to do help test a ROS2 image?

Also, is this all X86_64? Or can this be used with ARM CPU?

Thank you!

**Comment by at-wat** [comment]
Hi @tlalexander! Thank you for having interest to Alpine ROS project!

### ROS2

SEQSENSE Inc currently doesn't have specific plan to add ROS2 support to alpine-ros. If anyone could help, it's really welcome!
The things to do would be:
- Port dependencies if not provided in Alpine official repo (https://github.com/seqsense/aports-ros-experimental/tree/master/backports, it's on SEQSENSE's org since it uses the company's computation infra)
- Add buildtool (ament/colcon?) support to https://github.com/alpine-ros/ros-abuild-docker/blob/master/generate_rospkg_apkbuild/genapkbuild.py
- Add ROS2 distro to the buildfarm setting (https://github.com/seqsense/aports-ros-experimental/blob/master/.aports.yml)
- List ROS2 packages to be fetched from official rosdistro to https://github.com/seqsense/aports-ros-updater/ (some modification of the script would be needed)

## ARM Arch

We don't have plan for this as well.
Only x86_64 is supported on our build system at now, but I believe most of the builder scripts should work on ARM architecture as well.
Building https://github.com/seqsense/aports-ros-experimental with docker on ARM environment should generate packages for ARM.

**Comment by sequoia-hope** [comment]
Makes sense, thank you for the details! I am still figuring out how I want to build this. I will follow up here if I'm able to contribute!

**Comment by sequoia-hope** [comment]
Well, for anyone who finds this page I did sort out how to do ROS2 in Docker on Raspberry Pi (sadly not alpine).

You can see here that OSRF officially supports ROS2 on arm64 bit on Ubuntu Focal as a "Tier 1" support level, so it has full binaries and packages pre-built.
https://www.ros.org/reps/rep-2000.html#galactic-geochelone-may-2021-november-2022

So you need to use the Raspberry Pi 64 bit image, available here:
https://downloads.raspberrypi.org/raspios_arm64/images/

Once you install docker you will be able to build the ROS2 dockerfile in just a few minutes, since the packages are prebuilt.
https://github.com/osrf/docker_images/blob/master/ros/galactic/ubuntu/focal/ros-core/Dockerfile

And that's it!

You may want to close the issue, but since open issues are more visible to newbies I will leave it open and let you decide. Perhaps these instructions will inspire someone to contribute to a build for alpine.

On my Raspberry Pi, the base image for Alpine 3.12 is 5.34 MB, Ubuntu Focal is 65.6MB, and the ROS2 docker image for "ROS Base" [1] is 625MB.
[1] https://www.ros.org/reps/rep-2001.html#galactic-geochelone-may-2021-november-2022

Cheers!

**Comment by patrickelectric** [comment]
+1 for alpine and raspberry! armv7 and aarch64
